build/dockerfile.bc-saas

```diff
@@ -1,5 +1,5 @@
-ARG ARCH
 FROM golang:1.20.2
+ARG ARCH=amd64
```

Comment:
Thought we should set `ARCH` during build  at https://github.com/bestchains/bc-saas/blob/main/hack/lib/docker.sh#L38

which should
1. pass `ARCH` and `OS` to build script at Makefile

```shell
	@REGISTRY=$(REGISTRY) OUTPUT_TYPE=$(OUTPUT_TYPE) BUILD_PLATFORMS=$(GOOS)/$(GOARCH)  OS=$(GOOS) ARCH=$(GOARCH)  hack/build-image.sh $(filter ${DOCKER_TARGETS}, ${WHAT})
```


2. pass ARCH AND OS in build script `hack/lib/docker.sh`
```shell
		docker buildx build --progress=plain --output=type=$output_type \
			--platform "${BUILD_PLATFORMS}" \
			--build-arg ARCH="${ARCH}" \
			--build-arg OS="${OS}" \
			--tag "${image_name}" \
			--file "${ROOT_PATH}/build/dockerfile.${target}" \
			"${ROOT_PATH}"
```
3. Set `OS` and `ARCH` in dockerfile by default
```shell
ARG ARCH=amd64
ARG OS=linux
FROM golang:1.20.2
WORKDIR /go/src/github.com/bestchains/bc-saas
COPY . .
RUN make binary WHAT=depository GOARCH=${ARCH} GOOS=$(OS)
RUN make binary WHAT=market GOARCH=${ARCH} GOOS=$(OS)

FROM alpine:3.16
COPY --from=0 /go/src/github.com/bestchains/bc-saas/_output/bin/linux/${ARCH}/* /bin/*
WORKDIR /bin
```

Reply:
Root cause is that we forget to pass `ARCH` to the dockerfile